src/components/input/input.js

```diff
@@ -679,16 +694,17 @@ function mdMaxlengthDirective($animate, $mdUtil) {
       });
     });
 
-    function renderCharCount(value) {
-      // If we have not been initialized or appended to the body yet; do not render
+    function renderCharCount() {
+      // If we have not been initialized or appended to the body yet; do not render.
       if (!charCountEl || !charCountEl.parent) {
-        return value;
+        return;
       }
 
+      var elementVal = element.val();
+      elementVal = ngTrim && angular.isString(elementVal) ? elementVal.trim() : elementVal;
       // Force the value into a string since it may be a number,
       // which does not have a length property.
-      charCountEl.text(String(element.val() || value || '').length + ' / ' + maxlength);
-      return value;
+      charCountEl.text(String(elementVal || '').length + ' / ' + maxlength);
```

Comment:
If the intention is to have this working for numeric inputs as well (which might be the case, since [this comment][1] was removed), it will not work correctly for `0` :grin:

[1]: https://github.com/angular/material/pull/11136/files#diff-da06475ed96460fbcb944d6f1f31748aL193